### PR TITLE
Implement revert tests for VestingWallet

### PR DESCRIPTION
**Pull request by RenanSouza2**
Implement revert tests for VestingWallet

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

**Review by Amxx — DISMISSED** [review]
LGTM

**Review by Amxx — APPROVED** [review]

File: test/finance/VestingWallet.test.js [hunk 1]
```diff
@@ -62,11 +71,19 @@ describe('VestingWallet', function () {
         this.token = await ethers.deployContract('$ERC20', ['Name', 'Symbol']);
         await this.token.$_mint(this.mock, this.amount);
 
-        this.getBalance = account => this.token.balanceOf(account);
```
Comment (ernestognw) [inline test/finance/VestingWallet.test.js:65]: We don't need this functions anymore now that we're using `changeTokenBalances` and `changeEtherBalances`

**Review by ernestognw — APPROVED** [review]

**Comment by RenanSouza2** [comment]
@ernestognw I made a bigger cleanup, see if you approve it

**Comment by ernestognw** [comment]
> @ernestognw I made a bigger cleanup, see if you approve it

Thanks, I changed a couple things, the main one is that I brought back the `releasedEvent` and `argsVerify`. Otherwise, we're not checking the `ERC20Released` event. Also, I think the vesting wallet behavior should check for a release event every time, regardless what other checks are contained in `checkRelease`.

**Review by ernestognw — APPROVED** [review]

**Comment by RenanSouza2** [comment]
and what do you say abbout this proposal?

https://github.com/RenanSouza2/openzeppelin-contracts/tree/add-finance-tests-proposal/test/finance

**Comment by ernestognw** [comment]
> and what do you say abbout this proposal?
> 
> 
> 
> https://github.com/RenanSouza2/openzeppelin-contracts/tree/add-finance-tests-proposal/test/finance
> 
> 
> 
> I've incorporated both events tests in 'checkRelease'

Not convinced. Right now it makes no difference because the behavioral tests are only used here, but having the event checks outside the behavioral test seems an antipattern to me. It shouldn't be possible to pass the behavioral test without an event.

**Comment by RenanSouza2** [comment]
> Not convinced. Right now it makes no difference because the behavioral tests are only used here, but having the event checks outside the behavioral test seems an antipattern to me. It shouldn't be possible to pass the behavioral test without an event being emitted on releasing.

Makes sense, this is a way to enfor the verification of at least one test

**Review by ernestognw — APPROVED** [review]
Noted some small improvements. Re approving.

**Comment by ernestognw** [comment]
> Makes sense, this is a way to enforce the verification of at least one test

Yeah, I think it's a small decision honestly, so I'm open to changing it if @Amxx sees something I'm ignoring.

The PR is in good shape for merging in my opinion but we can create another PR if anyone feels strongly about one of the last decisions. I'm merging

**Pull request merged**